lib/sycamore/sycamore/connectors/file/file_scan.py

```diff
@@ -321,5 +363,42 @@ def execute(self, **kwargs) -> "Dataset":
         )
         return ds.flat_map(self.json_as_document, **self.resource_args)
 
+    def local_source(self, **kwargs) -> list[Document]:
+        if isinstance(self._paths, str):
+            paths = [self._paths]
+        else:
+            paths = self._paths
+
+        documents = []
+
+        def process_file(info):
```

Comment:
I know the actual document parsing is somewhat different between the two json methods, but can we factor out the filesystem stuff? Seems common with `BinaryScan.local_source` as well, no?

Reply:
yes, make sense, would try to factor out as much as possible.